src/extras/MovieClip.js

```diff
@@ -123,7 +151,12 @@ Object.defineProperties(MovieClip.prototype, {
     currentFrame: {
         get: function ()
         {
-            return Math.floor(this._currentTime) % this._textures.length;
+            var currentFrame = Math.floor(this._currentTime) % this._textures.length;
+            if (currentFrame < 0)
```

Comment:
It just handles the case that currentTime is negative (this is the case if animationSpeed (or deltaTime). e.g if currentTime is -4 and your animation has 3 frames it returns: (-4 % 3) + 3 = 2